Pod/Classes/IPStringBackedDecimalValue.m

```diff
@@ -0,0 +1,111 @@
+//
+//  IPStringBackedDecimalValue.m
+//  Pods
+//
+//  Created by Nicholas Servidio on 10/26/15.
+//
+//
+
+#import "IPStringBackedDecimalValue.h"
+#import "NSNumberFormatter+IPCurrencySansSymbol.h"
+
+NSUInteger const kIPDefaultMaxNumberOfWholeNumberDigits = 3;
+NSUInteger const kIPDefaultMaxNumberOfDecimalDigits = 2;
+
+@interface IPStringBackedDecimalValue ()
+
+@property (strong, nonatomic, readwrite) NSMutableString *currentValue;
+
+@end
+
+@implementation IPStringBackedDecimalValue
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        _currentValue = [@"" mutableCopy];
+        _maxNumberOfWholeNumberDigits = kIPDefaultMaxNumberOfWholeNumberDigits;
+        _maxNumberOfDecimalDigits = kIPDefaultMaxNumberOfDecimalDigits;
+    }
+    return self;
+}
+
+- (NSString *)currentValueFormattedAsPrice {
+    return [NSNumberFormatter valueFormattedAsPriceWithNoSymbol:[self.currentValue floatValue]];
+}
+
+- (void)addDigitToCurrentValue:(NSUInteger)digit {
+    if ([self canAddDigitToCurrentValue:digit]) {
+        [self.currentValue appendString:[NSString stringWithFormat:@"%ld", digit]];
+    }
+}
+
+- (void)addDecimalPointToCurrentValue {
+    if ([self canAddDecimalPointToCurrentValue]) {
+        if (self.currentValue.length == 0) {
+            [self addDigitToCurrentValue:0];
+        }
+        [self.currentValue appendString:@"."];
```

Comment:
Some locales format "." as the thousand separator. But this is probably fine for most of our situations.